app/models/coronavirus_restriction_search.rb

```diff
@@ -38,19 +35,54 @@ def invalid_postcode?
     %w[invalidPostcodeFormat fullPostcodeNoMapitValidation].include?(error_code)
   end
 
-  def no_restriction?
-    location_lookup.data.first&.england? && local_restriction.nil?
+  def devolved_nation?
+    ["Northern Ireland", "Scotland", "Wales"].include?(locations&.first&.country)
```

Comment:
We have helper methods for the country in MapitLocation, e.g. `northern_ireland?` etc. Can they be used instead. e.g

```
location = locations&.first
location.wales? || location.scotland? || location.northern_ireland?
```

Or 

```
!locations&.first&.england?
```